src/components/virtual-scroll/virtual-scroll.ts

```diff
@@ -368,7 +369,13 @@ export class VirtualScroll implements DoCheck, AfterContentInit, OnDestroy {
   /**
    * @input {function} Same as `ngForTrackBy` which can be used on `ngFor`.
    */
-  @Input() virtualTrackBy: TrackByFn;
+  @Input() set virtualTrackBy(val: TrackByFn) {
+    if (!isPresent(val)) return;
+    this._virtualTrackBy = val;
+    if (isPresent(this._records)) {
+      this._differ = this._iterableDiffers.find(this._records).create(this.virtualTrackBy);
```

Comment:
Wasn't sure if you prefer I overwrite last commit or add a new one, so I went with the safer choice.